pkg/netceptor/netceptor.go

```diff
@@ -1937,13 +1937,17 @@ func (s *Netceptor) runProtocol(ctx context.Context, sess BackendSession, bi *Ba
 						remoteCost, ok := ri.Connections[s.nodeID]
 						if !ok {
 							if remoteEstablished {
+								s.removeConnection(remoteNodeID)
```

Comment:
Do we actually want to do this here? Does this mean that a duplicate node will kick off the originally connected one?

Reply:
So I read this code as the remote node thinks it is no longer connected to the current node, however the current node still has the remote node under its connections and thinks it is actively connected.

As a result, the connection needs to be re-established and the connection table cleaned up.